ddtrace/tracer/option.go

```diff
@@ -343,7 +356,13 @@ func newConfig(opts ...StartOption) *config {
 		}
 		c.dogstatsdAddr = addr
 	}
-
+	if testSessionToken := os.Getenv("CI_TEST_AGENT_SESSION_TOKEN"); testSessionToken != "" {
+		tr, ok := c.transport.(*httpTransport)
+		if ok {
+			tr.headers["X-Datadog-Trace-Env-Variables"] = c.DumpForTestAgent()
+			tr.headers["X-Datadog-Test-Session-Token"] = testSessionToken
+		}
+	}
```

Comment:
You should provide your own [httpClient](https://github.com/DataDog/dd-trace-go/blob/09f5ea49c29ebc547a6d0054ccf85cff5b861ba2/ddtrace/tracer/option.go#L708) when starting the tracer in the test instead of hardcoding this very custom logic.

Reply:
done